Bio/PDB/PDBList.py

```diff
@@ -112,6 +139,27 @@ def __init__(
         # variable for command-line option
         self.flat_tree = False
 
+    def _get_pdb_server_url(self, server: str | PDBServer | None) -> str:
+        """Get remote PDB server url.
+
+        Handle legacy server declaration (str).
+        If no server specified, chose the fastest one.
+        """
+        pdb_server = None
+        if server is None:
+            pdb_server = get_fastest_server()
+        elif isinstance(server, str):
+            return urljoin(server, "pub/pdb/")
+        elif isinstance(server, PDBServer):
+            pdb_server = server
+
+        if pdb_server:
+            return urlunsplit(
+                ("ftp", pdb_server.domain, pdb_server.pdb_directory, None, None)
+            )
+
+        raise TypeError(f"Unexpected server (server: {server}).")
```

Comment:
Suggestion: `TypeError(f"Unsupported server option: {server}")`

Reply:
Why the redudant `server argument server: {server}` and not just `server argument :{server}`?

Reply:
It is a good practice that I have taken over the years, formatting log and error message as `Message (vars).`. 
In this specific and isolated case, it's hard to see the benefits, but when we have a (lot) of long text with tones of vars inside, it makes a big difference. This way all messages are uniform and readable and the log search is much easier.

But I'm not going to fight for this one, if you insist I can implement your suggestion as is.

Reply:
I'd insist on this one, it's adding verbosity without adding any more information: `"Unexpected value for server argument: {server}"` is informative and concise.

Reply:
```suggestion
        raise TypeError(f"Unsupported server option: {server}")
```